### ISSUE TITLE
mouse keys and massdrop configurator?

### ISSUE THREAD

**Issue by inkthink** [ErgoDox-EZ/docs#2]
Are the mouse keys or hyper keys available with the massdrop configurator or does that require C?
Also, will using the massdrop configuration tool to alter the base layer override the special layers that are included by default?

**Comment by ezuk** [comment]
That does require C. And yes, using the massdrop configurator will override the entire firmware -- nothing is left behind.

Are you familiar with the firmware repository at https://github.com/jackhumbert/qmk_firmware/tree/master/keyboard/ergodox_ez ? Getting started is not as daunting as it seems!

**Comment by inkthink** [comment]
Yes, I've been pouring over the configurators for a couple of weeks, as time allows. 
I was glad to find that [keymap_default.c](https://github.com/jackhumbert/qmk_firmware/blob/master/keyboard/ergodox_ez/keymaps/keymap_default.c) and the code seems straightforward at first glance. 
I was getting ready to try forking it, actually. 
I got as far as looking up step three [gcc compiler](https://www.pjrc.com/teensy/gcc.html) and discovering the Mac download (AVR MacPack) link is broken. 
I haven't yet gotten to hunt for alternatives, but it got me wondering if I might get a much faster start with the massdrop configurator. My customized Kinesis layout for the thumb cluster is firmly memorized and I'm reluctant to retrain temporarily so the EZ's been sitting mostly unused. 

P.S. Thanks for the encouragement to try it. I plan to eventually, even if I do throw something basic together with massdrop for now.

**Comment by sethherr** [comment]
@inkthink - you should use homebrew to compile avr

It's easy:

```
brew tap osx-cross/avr
brew install avr-libc
```

Then you can just cd into the ergodox directory and

```
make clean
make KEYMAP="yourkeymapname"
```

**Comment by inkthink** [comment]
Thanks, that's what I was looking for. I'm looking forward to using it. Meanwhile i installed a temporary/testing layout using the massdrop configurator. That was very easy. I did not use the eep file mentioned in these docs, since the link was broken, and the qmk repo doesn't mention it. It worked just fine without it. i'm looking forward to improving my layout with code, but at least now I'm using my keyboard... literally now, to type this. :-)

**Comment by sethherr** [comment]
Thanks for the heads up about the firmware.eep broken link. It's been fixed

**Comment by Eric-L-T** [comment]
Hello, World!

**Comment by ezuk** [comment]
Okay, this worked. Interesting...

**Comment by Eric-L-T** [comment]
@ezuk Do you have any idea what might be wrong?

**Comment by ezuk** [comment]
I don't, actually. I can't access repo settings for qmk, so I'm just not
sure. I suggest checking with GitHub support?

**Comment by inkthink** [comment]
Testing my first full layout now. I find the mouse keys handy... since my mouse broke. I also use caps lock sometimes, which I have mapped to layer 1. I would like to incorporate LED's and found the following instructions for it, but I'm not sure where to place it in the keymap:

```
if (host_keyboard_leds() & (1<<USB_LED_CAPS_LOCK)) {
  ergodox_right_led_3_on();
}
```

**Comment by ezuk** [comment]
That looks like a macro - I suggest opening an issue over at [the qmk repo](https://github.com/jackhumbert/qmk_firmware) with this question, lots of action there :)

**Comment by eltang** [comment]
@inkthink Take a look at the `matrix_scan_user` function in the default ErgoDox EZ keymap!
